code/src/java/translation/util/Tips.java

```diff
@@ -340,16 +361,28 @@ public static void generateTips(File rootDirectory, File translation, String tra
 									String translatedLine = tipsTranslated.get(readLine);
 									if (translatedLine == null)
 									{
-										log("null translated line in {1}, original {0}", readLine, translation);
+										log(
```

Comment:
While you're in this file, can you see if there is a sane way to break up this method?  This is like 8 indents in, which is a bit silly